src/plugins/MediaManager.js

```diff
@@ -0,0 +1,339 @@
+'use strict';
+
+const log = require('loglevel');
+log.setDefaultLevel('debug');
+
+module.exports = class MediaManager {
+    /**
+     * Constructor to this MediaManager class
+     * @param {DeviceRenderer} instance root instance
+     * @param {boolean} videoWithMicrophone flag indicating wether the video must always be played with the microphone. Defaults to the microphone flag in the instance options
+     * @param {number} videoWidth Maximum video width. Defaults to 1280
+     * @param {number} videoHeight Maximum video height. Defaults to 720
+     */
+    constructor(instance, videoWithMicrophone = instance.options.microphone, videoWidth = 1280, videoHeight = 720) {
+        this.instance = instance;
+        this.instance.mediaManager = this;
+
+        if (!navigator.mediaDevices || !this.setupPermissions()) {
+            log.error('MediaDevices API unsupported: camera and microphone won\'t be available.');
+            this.instance.mediaEventsEnabled = false;
+            return;
+        }
+        this.instance.mediaEventsEnabled = true;
+
+        this.videoStreaming = false;
+        this.audioStreaming = false;
+
+        this.videoWidth = videoWidth;
+        this.videoHeight = videoHeight;
+        this.videoWithMicrophone = videoWithMicrophone;
+
+        this.setupPermissions();
```

Comment:
is it really necessary to redo a setupPermissions ? if [l.18](https://github.com/Genymobile/genymotion-device-web-player/pull/40/files#diff-97a1a105b0f7db226929204ef391737aac32f2a5cbaf8a20278a55453fb07cddR18) passes it should already have been done so redoing it might double register the `on<Microphone/Camera>PermissionChange` listeners

Reply:
You're right, I forgot to remove this line when adding l.18 :) Thanks!